lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +198,89 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self.full_url = full_url
+        self.article_id = article_id
+        self.config = config
+        self.article = Article(full_url, article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        article = article_soup.find_all('div', {'itemprop': 'articleBody'})[0]
+        article_list = article.find_all('p')
+        paragraphs = [par.text for par in article_list]
+        self.article.text = '\n'.join(paragraphs)
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        article = article_soup.find_all('div', {'itemprop': 'headline'})[0]
+        article_title = article.find_all('h1')[0]
+        self.article.title = article_title.text
+        article_authors = article_soup.find_all('span', {'itemprop': 'author'})[0]
+        authors = article_authors.find_all('span', {'itemprop': 'name'})
+        if authors:
+            self.article.author = [author.text for author in authors]
+        else:
+            self.article.author = ['NOT FOUND']
+        try:
+            article_tags = article_soup.find_all('ul', {'itemprop': 'keywords'})[0]
+            article_tags_li = article_tags.find_all('li')
+            self.article.topics = [tag.find_all('a')[0].text for tag in article_tags_li]
+        except IndexError:
+            self.article.topics = []
+        article_date = article_soup.find_all('span', {'class': 'news-datetime mb-10 mr-20'})[0]
+        self.article.date = self.unify_date_format(article_date.text)
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        date_list = date_str.split()
+        date_list[1] = date_list[1][:3]
+        if date_list[1] == 'мая':
+            date_list[1] = 'май'
+        date_str = ' '.join(date_list)
+        locale.setlocale(category=locale.LC_ALL, locale='Russian')
+        return datetime.datetime.strptime(date_str, '%d %b %Y г. %H:%M')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = requests.get(self.full_url)
+        bs = BeautifulSoup(response.text, 'lxml')
+        self._fill_article_with_text(bs)
+        self._fill_article_with_meta_information(bs)
+        return self.article
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    new_path = Path(__file__).parent / base_path
+    new_path.mkdir(exist_ok=True, parents=True)
```

Comment:
exist ok is unnecessary as you just removed it

Reply:
Done